sympy/core/tests/test_function.py

```diff
@@ -239,7 +238,9 @@ def test_Lambda():
     p = x, y, z, t
     assert Lambda(p, t*(x + y + z))(*p) == t * (x + y + z)
 
-    assert Lambda(x, 2*x) + Lambda(y, 2*y) == 2*Lambda(x, 2*x)
+    eq = Lambda(x, 2*x) + Lambda(y, 2*y)
+    assert eq != 2*Lambda(x, 2*x)
```

Comment:
Looking at this I suddenly realise the `Lambda` is a subclass of `Expr`. That should be changed...

Reply:
Why? Sum is an Expr, too, and 
```python
>>> eq = Sum(x, (x, 1, 2))
>>> eq + eq.xreplace({x:y})
Sum(x, (x, 1, 2)) + Sum(y, (y, 1, 2))
```

Reply:
I'll open a separate issue but just on that point: A Sum represents an expression e.g. both sums above are in fact integers. A function is not anything like a number. The application of the function can give an expression but the function itself is a "callable". Also it's not necessarily the case that a Lambda even evaluates to an Expr:
```julia
In [22]: f = Lambda(x, (x, x))

In [23]: f(1)
Out[23]: (1, 1)
```

Reply:
Reading this now, I am guessing that you mean that Lambda addition should be prohibited like Eq addition.